### PR TITLE
Slack contrib for Docker

### PR DESCRIPTION
**Pull request by anddann**
Addresses issue #12

### CODE REVIEW

**Comment by anddann** [comment]
This PR needs a litte bit more love, since manipulating files using `sed` and especially the json file `conf.json` is a bit hacky. However, I don't have a better idea at the moment.

**Comment by ajira86** [comment]
I propose to create a "script" folder to put checkdb.py and other script.
The idea is to keep root clear with only docker-entrypoint.sh and welcome instructions (README, CONTRIBUTING ...)

**Comment by anddann** [comment]
Hi @ajira86,
I've an idea how the get ride of the "ugly" and brittle sed commands in the 'docker-entrypoint.sh' and my 'slack' script.
The container installs taiga, taiga depends on python, and python always includes json libraries. Thus, we can easily replace the 'sed' commands with a simple python scripts to change properties in the file `/taiga/conf.json`

**Comment by anddann** [comment]
@ajira86 Your idea for a separate `scripts` folder is excellent.
I've included it in this PR.

**Comment by anddann** [comment]
And I've migrated to script code to a python script to make it more robust.
Would you please review these changes?

**Comment by ajira86** [comment]
I'm just a little bit short in time to review your PR right now. But I'll try to do it this weekend.

File: scripts/config-slack-plugin.sh [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+   # activate in the fronted
+  if grep -Fxq "$TAGIA_SLACK_CONFIG_FRONTED" "$TAIGA_CONFIG_FILE_FRONTEND"
+  then
+      # frontend is configured
+      echo "Slack already actived in fronted"
+  else
+      # frontend is not  configured
+      echo "Activate Slack in fronted"
+      PYTHON_CMD='import modify_conf; modify_conf.modifyJSONFile("/taiga/conf.json","contribPlugins","'"$TAIGA_CONFIG_FILE_FRONTEND"'")'
```
Comment (ajira86) [inline scripts/config-slack-plugin.sh]: shouldn't be this value $TAGIA_SLACK_CONFIG_FRONTED ?

File: scripts/config-slack-plugin.sh [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+  if grep -Fxq "$TAGIA_SLACK_CONFIG_FRONTED" "$TAIGA_CONFIG_FILE_FRONTEND"
+  then
+      # frontend is configured
+      echo "Deactivate Slack in fronted"
+      PYTHON_CMD='import modify_conf; modify_conf.modifyJSONFile("/taiga/conf.json","contribPlugins","'"$TAIGA_CONFIG_FILE_FRONTEND"'",True)'
+      cd /scripts/ && python -c "$PYTHON_CMD"
+      cd $WORKINGDIR
+      echo "Slack deactived in fronted"
+  else
+  # TODO: add code to deactivate in frontend
```
Comment (ajira86) [inline scripts/config-slack-plugin.sh]: Is this TODO still necessary ?

File: config.env [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# TAIGA_EMAIL_FROM=no-reply@taiga.mycompany.net
+# TAIGA_EMAIL_USE_TLS=True
+# TAIGA_EMAIL_HOST=smtp.google.com
+# TAIGA_EMAIL_PORT=587
+# TAIGA_EMAIL_USER=me@gmail.com
+# TAIGA_EMAIL_PASS=super-secure-pass phrase thing!
+
+
+# uncomment for slack
+#TAIGA_SLACK=ACTIVE
```
Comment (ajira86) [inline config.env]: Isn't it better to use something like SLACK_ACTIVE=True to be more consistent ?

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -27,29 +29,39 @@ fi
...
-  echo "Disabling SSL support!"
-  sed -i "s#https://#http://#g" /taiga/conf.json
-  sed -i "s#wss://#ws://#g" /taiga/conf.json
-fi
 
 # Reinitialize nginx links
-rm /etc/nginx/sites-enabled/*
-if [ "$TAIGA_SSL_BY_REVERSE_PROXY" = "True" ] || [ "$TAIGA_SSL" = "True" ]; then
+if [ "$TAIGA_SSL" = "True" ]; then
+  rm /etc/nginx/sites-enabled/*
```
Comment (ajira86) [inline docker-entrypoint.sh]: Why to reinitialize links for SSL only what happens with taiga-events in this case ?

File: scripts/config-slack-plugin.sh (hunk 2 above)
Comment (anddann) [reply scripts/config-slack-plugin.sh]: nope. deleted.

File: scripts/config-slack-plugin.sh (hunk 1 above)
Comment (anddann) [reply scripts/config-slack-plugin.sh]: Thanks! You are right. Fixed.

File: config.env (hunk 3 above)
Comment (anddann) [reply config.env]: Yeah, its easier if we keep it consistent. Fixed.

File: docker-entrypoint.sh (hunk 4 above)
Comment (anddann) [reply docker-entrypoint.sh]: Nope, as discussed in #14 , this was a mistake. Fixed

**Comment by anddann** [comment]
@ajira86 
If we want to setup more plugins, we do not need to refactor this but create further `sh` scripts.
This slack script may serve as a template for further plugins.

**Pull request merged**